### PR TITLE
fix(Script/Creature): Move Shadow Hunter Vosh'gajin script to SAI

### PR DESCRIPTION
**Pull request by wizzymore**
## Changes Proposed:
- Moves the script from C++ to SAI
- Fixes the NPC script by removing cleave and puttin the correct spell id's to Hex and Curse of Blood
- Fix Hex to have the correct 1.5 cast time ( was instant )

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7363

## SOURCE:

https://www.wowhead.com/spell=16097/hex
https://www.wowhead.com/npc=9236/shadow-hunter-voshgajin#abilities;mode:normal
https://www.youtube.com/watch?v=3bKwDcQqINM&t=648s

## Tests Performed:

- Re-builded the core
- Applied the SQL
- Tested by killing the NPC and everything works as intended.

## How to Test the Changes:

1. Apply the SQL, rebuild the core to have the C++ script removed and the fix to the Hex spell
2. .go c 43522

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@wizzymore please fix the merge conflicts and we will merge this

**Comment by Kitzunu** [comment]
Conflict

**Comment by wizzymore** [comment]
Hi guys, i cannot fix any conflicts currently, i am away from my home and have no laptop with me and when trying to fix it in the browser it just crashes all the time.

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -7277,6 +7277,11 @@ void SpellMgr::LoadDbcDataCorrections()
         spellInfo->AttributesEx |= SPELL_ATTR1_NO_THREAT;
     });
 
+    // 16097 Shadow Hunter Vosh'gajin - Hex
+    ApplySpellFix({ 16097 }, [](SpellEntry* spellInfo) {
+        spellInfo->CastingTimeIndex = 16;
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellMgr.cpp:7282]: put the bracket on a new line, follow how it has been done before please =)

**Comment by Kitzunu** [comment]
fixed the conflict

**Pull request merged**